src/main/java/edu/garyMooradian/advancedJava/StockServiceFactory.java

```diff
@@ -0,0 +1,29 @@
+package edu.garyMooradian.advancedJava;
+
+import org.apache.http.annotation.Immutable;
+
+/*
+ * This class will return an instantiation of a specific StockService (i.e. BasicStockService)
+ * BasicStockService implements StockService, therefore, we can return
+ * BasicStockService as StockService type
+ */
+@Immutable
+public class StockServiceFactory {
+	
+	/*
+	 * We call this method getStockService because it returns any implementation
+	 * of StockService; i.e. we don't want to specify a particular implementation
```

Comment:
carefully commenting code is important and I appreciate the effort.. And while it is better to have too much comments, this entry is too verbose.